pycqed/measurement/sweep_points.py

```diff
@@ -345,3 +348,30 @@ def find_parameter(self, param_name):
             if param_name in self[dim]:
                 return dim
         return None
+
+    def subset(self, i, dimension=0):
+        """
+        Returns a new SweepPoints object with one of the dimensions reduced
+        to a subset of the sweep values. The other dimensions are unchanged.
+        :param i: (list) indices of the sweep values that should be
+            contained in the subset.
+        :param dimension: (int, default 0) index of the dimension that
+            should be reduced
+        """
+        sp = SweepPoints(self)
+        for k, v in sp[dimension].items():
+            sp[dimension][k] = (np.array(v[0])[i], v[1], v[2])
+        return sp
```

Comment:
Should we use the `get_sweep_param_property` here instead of explicit indexing of `v`? Or you think it's overkill?

Reply:
I guess the purpose of get_sweep_param_property  is that you do not need to refactor other modules in case you ever decide to change the internal storage structure in SweepPoints. But if you ever decide to do this, you have to refactor code in the sweep_points module anyways, and it should be fine to also refactor these two lines of code. If you just use get_sweep_param_property, you still need to store it in the correct format afterwards. So probably you would need a combination of get_sweep_param_property and add_sweep_parameter to really get it independent of the storage format. Or do you have a clever solution for this?

Reply:
Yeah ok that is true. I do not have a solution for it, but like you say we would need to update this module if the structure of SweepPoints changes, so then it's fine.